src/subsystems/racket/translators/racket-impure/generate-dream-lock.rkt

```diff
@@ -0,0 +1,150 @@
+           [generic (make-immutable-hash
+                     `((subsystem . "racket")
+                       (location . ,rel-path)
+                       (sourcesAggregatedHash . ,(json-null))
+                       (defaultPackage . ,package-name)
+                       (packages . ,(make-immutable-hash `((,(string->symbol package-name) . "0.0.0"))))))]
+           [sources-from-catalog
+            (hash-map pkgs-all
+                      (match-lambda**
+                       [(name (hash-table
+                               ('versions
+                                (hash-table
+                                 ('default
+                                  (hash-table
+                                   ('source_url url)))))
+                               ('checksum rev)))
+                        (let* ([source-with-removed-http-or-git-double-slash (regexp-replace #rx"^(?:git|http)://" url "https://")]
+                               [left-trimmed-source (string-trim source-with-removed-http-or-git-double-slash "git+" #:right? #f)]
+                               [maybe-match-path (regexp-match #rx"\\?path=([^#]+)" left-trimmed-source)]
+                               [trimmed-source (regexp-replace #rx"(?:/tree/.+)?(?:\\?path=.+)?$" left-trimmed-source "")])
+                          (cons (string->symbol name)
+                                (make-immutable-hash
+                                 `((0.0.0 . ,(make-immutable-hash
+                                              (append (match maybe-match-path
+                                                        [(list _match dir)
+                                                         `((dir . ,(regexp-replace* #rx"%2F" dir "/")))]
+                                                        [_ '()])
+                                               `((url . ,trimmed-source)
+                                                 (rev . ,rev)
+                                                 (type . "git")
+                                                 ;; TODO: sha256?
+                                                 ))))))))]))]
+           [sources-from-repo (if (string=? rel-path "")
+                                  (list (cons (string->symbol package-name)
+                                              (make-immutable-hash
+                                               `((0.0.0 . ,(make-immutable-hash
+                                                            `((type . "path")
+                                                              (path . ,src-path))))))))
+                                  (set-map names-of-overridden-packages
+                                           (lambda (name)
+                                             (cons (string->symbol name)
+                                                   (make-immutable-hash
+                                                    `((0.0.0 . ,(make-immutable-hash
```

Comment:
The `version` property that people sometimes put in their `info.rkt` is essentially meaningless since every commit that occurs after version X is cut will replace update the repo with the new source but still keep it at version X.